instrumentation/kafka/kafka-clients/kafka-clients-0.11/javaagent/src/test/java/io/opentelemetry/instrumentation/kafkaclients/KafkaClientSuppressReceiveSpansTest.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafkaclients;
+
+import static io.opentelemetry.instrumentation.testing.util.TelemetryDataUtil.orderByRootSpanKind;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.satisfies;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.kafka.internal.KafkaClientBaseTest;
+import io.opentelemetry.instrumentation.kafka.internal.KafkaClientPropagationBaseTest;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.assertj.core.api.AbstractLongAssert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class KafkaClientSuppressReceiveSpansTest extends KafkaClientPropagationBaseTest {
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Test
+  public void testKafkaProduceAndConsume() throws InterruptedException {
+    String greeting = "Hello Kafka!";
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          producer.send(
+              new ProducerRecord<>(SHARED_TOPIC, greeting),
+              (meta, ex) -> {
+                if (ex == null) {
+                  testing.runWithSpan("producer callback", () -> {});
+                } else {
+                  testing.runWithSpan("producer exception: " + ex, () -> {});
+                }
+              });
+        });
+
+    awaitUntilConsumerIsReady();
+    // check that the message was received
+    ConsumerRecords<?, ?> records = consumer.poll(Duration.ofSeconds(5).toMillis());
+    for (ConsumerRecord<?, ?> record : records) {
+      testing.runWithSpan(
+          "processing",
+          () -> {
+            assertThat(record.value()).isEqualTo(greeting);
+            assertThat(record.key()).isNull();
+          });
+    }
+
+    testing.waitForTraces(1);
+    testing.waitAndAssertTraces(
```

Comment:
The `waitAndAssertTraces()` method already waits for 1 completed trace, there should be no need to call `waitForTraces()` explicitly
```suggestion
    testing.waitAndAssertTraces(
```

Reply:
removed most `testing.waitForTraces` but kept two where there is no `testing.waitAndAssertTraces`